04-OO_field_guide.Rmd

```diff
@@ -42,6 +42,50 @@ library(methods)
     methods(class = "POSIXlt")
     ```
 
+    Compare method behaviors between the two classes:
+   
+    ```{r}
+    
+    #get method information for the two classes
+    ct_ms=attr(methods(class='POSIXct'),'info')
+    lt_ms=attr(methods(class='POSIXlt'),'info')
+    
+    #get corresponding functions 
+    ct_fs=lapply(1:NROW(ct_ms),function(i){
+      x=ct_ms[i,]
+      if(x$isS4=='TRUE'){
+        a=rownames(ct_ms)[i]
+        sigs=strsplit(a,',|-method')[[1]][-1]
+        return(getMethod(x$generic,sigs))
```

Comment:
The book recommends to use explicit returns when there are multiple returns so that the code is more readable.  I didn't use return in the last sapply because there was only one return.